server/server/src/configuration.rs

```diff
@@ -119,12 +132,14 @@ pub fn copy_example_environment_configuration_file_if_required() -> Result<(), S
 ///
 /// Individual settings properties can be also manually defined as environment variables: see
 /// `get_configuration_environment` for details.
-pub fn get_configuration() -> Result<Settings, SettingsError> {
+pub fn get_configuration(config_file: Option<PathBuf>) -> Result<Settings, SettingsError> {
```

Comment:
Should this one get ConfigArgs rather then config_file ? Seems like you've made ConfigArgs, rathern then just config_file: Option<PathBuf>, so that future configuration additions would require less code change (since cli and main executeable share these configs), it feels natural in this sense to pass ConfigArgs to get_configuration